From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:14:12 +0530
Subject: [PATCH 034/109] Cygwin: Check if gawk is available in
 gentls_offsets script

(cherry picked from commit feeaf5f36acd0cfcacf0e001083bf5310e6a8a55)
---
 winsup/cygwin/scripts/gentls_offsets | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/winsup/cygwin/scripts/gentls_offsets b/winsup/cygwin/scripts/gentls_offsets
index c375a6106..bf84dd0cb 100755
--- a/winsup/cygwin/scripts/gentls_offsets
+++ b/winsup/cygwin/scripts/gentls_offsets
@@ -6,6 +6,12 @@ tmp_file=/tmp/${output_file}.$$
 
 trap "rm -f ${tmp_file}" 0 1 2 15
 
+# Check if gawk is available
+if ! command -v gawk &> /dev/null; then
+    echo "$0: gawk not found." >&2
+    exit 1
+fi
+
 # Preprocess cygtls.h and filter out only the member lines from
 # class _cygtls to generate an input file for the cross compiler
 # to generate the member offsets for tlsoffsets-$(target_cpu).h.
